diff --git a/dlls/d2d1/command_list.c b/dlls/d2d1/command_list.c
index 4d81f0f1ccf..a7f67f322af 100644
--- a/dlls/d2d1/command_list.c
+++ b/dlls/d2d1/command_list.c
@@ -475,8 +475,8 @@ static HRESULT STDMETHODCALLTYPE d2d_command_list_Close(ID2D1CommandList *iface)
 
     FIXME("iface %p stub.\n", iface);
 
-    if (command_list->state != D2D_COMMAND_LIST_STATE_OPEN)
-        return D2DERR_WRONG_STATE;
+    //if (command_list->state != D2D_COMMAND_LIST_STATE_OPEN)
+        //return D2DERR_WRONG_STATE;
 
     command_list->state = D2D_COMMAND_LIST_STATE_CLOSED;
 
