diff --git a/dlls/d2d1/effect.c b/dlls/d2d1/effect.c
index b66fe731523..95cdc2a84bc 100644
--- a/dlls/d2d1/effect.c
+++ b/dlls/d2d1/effect.c
@@ -17,7 +17,11 @@
  */
 
 #include "d2d1_private.h"
-
+#include "d2d1effects_2.h"
+//#undef E_INVALIDARG
+//#define E_INVALIDARG S_OK
+//#undef D2DERR_INVALID_PROPERTY
+//#define D2DERR_INVALID_PROPERTY S_OK
 WINE_DEFAULT_DEBUG_CHANNEL(d2d);
 
 static inline struct d2d_transform *impl_from_ID2D1OffsetTransform(ID2D1OffsetTransform *iface)
@@ -1468,6 +1472,52 @@ static HRESULT __stdcall gaussian_blur_factory(IUnknown **effect)
     return d2d_effect_create_impl(effect, &properties, sizeof(properties));
 }
 
+
+static const WCHAR histogram_description[] =
+L"<?xml version='1.0'?>                                                   \
+  <Effect>                                                                \
+    <Property name='DisplayName' type='string' value='Histogram'/>    \
+    <Property name='Author'      type='string' value='The Wine Project'/> \
+    <Property name='Category'    type='string' value='Stub'/>             \
+    <Property name='Description' type='string' value='Hisogram'/>    \
+    <Inputs>                                                              \
+      <Input name='Source'/>                                              \
+    </Inputs>                                                             \
+    <Property name='StandardDeviation' type='float' />                    \
+    <Property name='Optimization' type='enum' />                          \
+    <Property name='BorderMode' type='enum' />                            \
+  </Effect>";
+
+struct histogram_properties
+{
+    float standard_deviation;
+    D2D1_GAUSSIANBLUR_OPTIMIZATION optimization;
+    D2D1_BORDER_MODE border_mode;
+};
+
+EFFECT_PROPERTY_RW(histogram, standard_deviation, FLOAT)
+EFFECT_PROPERTY_RW(histogram, optimization, ENUM)
+EFFECT_PROPERTY_RW(histogram, border_mode, ENUM)
+
+static const D2D1_PROPERTY_BINDING histogram_bindings[] =
+{
+    { L"StandardDeviation", BINDING_RW(histogram, standard_deviation) },
+    { L"Optimization", BINDING_RW(histogram, optimization) },
+    { L"BorderMode", BINDING_RW(histogram, border_mode) },
+};
+
+static HRESULT __stdcall histogram_factory(IUnknown **effect)
+{
+    static const struct histogram_properties properties =
+    {
+        .standard_deviation = 3.0f,
+        .optimization = D2D1_GAUSSIANBLUR_OPTIMIZATION_BALANCED,
+        .border_mode = D2D1_BORDER_MODE_SOFT,
+    };
+    return d2d_effect_create_impl(effect, &properties, sizeof(properties));
+}
+
+
 static const WCHAR point_specular_description[] =
 L"<?xml version='1.0'?>                                                   \
   <Effect>                                                                \
@@ -1828,6 +1878,7 @@ void d2d_effects_init_builtins(struct d2d_factory *factory)
         { &CLSID_D2D1Flood, X2(flood) },
         { &CLSID_D2D1GaussianBlur, X2(gaussian_blur) },
         { &CLSID_D2D1PointSpecular, X2(point_specular) },
+        { &CLSID_D2D1Histogram, X2(histogram) },
         { &CLSID_D2D1ArithmeticComposite, X2(arithmetic_composite) },
         { &CLSID_D2D1Blend, X2(blend) },
         { &CLSID_D2D1Brightness, X2(brightness) },
@@ -3425,11 +3476,24 @@ HRESULT d2d_effect_create(struct d2d_device_context *context, const CLSID *effec
     UINT32 input_count;
     WCHAR clsidW[39];
     HRESULT hr;
+    ID2D1EffectImpl *impl = NULL; 
 
     if (!(reg = d2d_factory_get_registered_effect(context->factory, effect_id)))
     {
         WARN("Effect id %s not found.\n", wine_dbgstr_guid(effect_id));
-        return D2DERR_EFFECT_IS_NOT_REGISTERED;
+        if(IsEqualCLSID(effect_id, &CLSID_D2D1OpacityMetadata) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1Histogram) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1AlphaMask) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1ConvolveMatrix) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1Contrast) || 
+            IsEqualCLSID(effect_id, &CLSID_D2D1Opacity) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1BitmapSource) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D1Emboss) ||
+            IsEqualCLSID(effect_id, &CLSID_D2D12DAffineTransform)
+        )
+            reg = d2d_factory_get_registered_effect(context->factory, &CLSID_D2D1Blend);
+        else
+            return D2DERR_EFFECT_IS_NOT_REGISTERED;
     }
 
     if (!(effect_context = calloc(1, sizeof(*effect_context))))
@@ -3471,8 +3535,9 @@ HRESULT d2d_effect_create(struct d2d_device_context *context, const CLSID *effec
         ID2D1Effect_Release(&object->ID2D1Effect_iface);
         return hr;
     }
+    ID2D1EffectImpl_QueryInterface(object->impl, &IID_ID2D1EffectImpl, (void**)&impl );
 
-    if (FAILED(hr = ID2D1EffectImpl_Initialize(object->impl, &effect_context->ID2D1EffectContext_iface,
+    if (FAILED(hr = ID2D1EffectImpl_Initialize(impl, &effect_context->ID2D1EffectContext_iface,
             &object->graph->ID2D1TransformGraph_iface)))
     {
         WARN("Failed to initialize effect, hr %#lx.\n", hr);
